### PR TITLE
Add support for Zenvia WhatsApp

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: handlers/zenviawhatsapp/zenviawhatsapp.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	urn, err := urns.NewWhatsAppURN(payload.Message.From)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+
+	contactName := payload.Visitor.Name
+
+	msgs := []courier.Msg{}
+
+	for i := 0; i < len(payload.Message.Contents); i++ {
```
Comment (nicpottier) [inline handlers/zenviawhatsapp/zenviawhatsapp.go:111]: why not `range`?

**Review by nicpottier — APPROVED** [review]
Do we have a number we are testing this with?

**Comment by norkans7** [comment]
No, we do not have a number yet a customer wanted that so he will probably help test that as beta channel

I just followed the API docs

**Review by ntwarijoshua — APPROVED** [review]
LGTM. IMO, I would avoid init and use provider functions as those are easier to test

**Comment by norkans7** [comment]
> LGTM. IMO, I would avoid init and use provider functions as those are easier to test

I will read up more on that and see

**Comment by nicpottier** [comment]
The `init` pattern here is basically registering the type as a factory. That keeps the concerns of what types are enabled as easy as what you import in your `main`, which we have found to work well. I don't think a provider is going to do anything different there.

**Pull request merged**